plugin/storage/es/spanstore/reader.go

```diff
@@ -135,24 +136,45 @@ type timeRangeIndexFn func(indexName string, startTime time.Time, endTime time.T
 
 type sourceFn func(query elastic.Query, nextTime uint64) *elastic.SearchSource
 
-func getTimeRangeIndexFn(archive, useReadWriteAliases bool) timeRangeIndexFn {
+func getTimeRangeIndexFn(archive, useReadWriteAliases bool, remoteReadClusters []string) timeRangeIndexFn {
 	if archive {
 		var archivePrefix string
 		if useReadWriteAliases {
 			archivePrefix = archiveReadIndexSuffix
 		} else {
 			archivePrefix = archiveIndexSuffix
 		}
-		return func(indexName string, startTime time.Time, endTime time.Time) []string {
+		return addRemoteReadClusters(func(indexName string, startTime time.Time, endTime time.Time) []string {
 			return []string{archiveIndex(indexName, archivePrefix)}
-		}
+		}, remoteReadClusters)
 	}
+
 	if useReadWriteAliases {
-		return func(indices string, startTime time.Time, endTime time.Time) []string {
+		return addRemoteReadClusters(func(indices string, startTime time.Time, endTime time.Time) []string {
 			return []string{indices + "read"}
+		}, remoteReadClusters)
+	}
+
+	return addRemoteReadClusters(timeRangeIndices, remoteReadClusters)
+}
+
+// Elasticsearch cross cluster query api example: GET /twitter,cluster_one:twitter,cluster_two:twitter/_search
+// Add a remote cluster prefix for each cluster and for each index and add it to the list of original indices
+func addRemoteReadClusters(fn timeRangeIndexFn, remoteReadClusters []string) timeRangeIndexFn {
+	return func(indices string, startTime time.Time, endTime time.Time) []string {
```

Comment:
shouldn't this be short-circuited to return `fn` if `len(remoteReadClusters) == 0`?